doozerlib/olm/bundle.py

```diff
@@ -0,0 +1,450 @@
+        )[0]['digest']
+
+    def append_related_images_spec(self, contents, images):
+        """Create a new section under contents' "spec" called "relatedImages", listing all given
+        images in the following format:
+
+        spec:
+          relatedImages:
+            - name: image-a
+              image: registry/image-a@sha256:....
+            - name: image-b
+              image: registry/image-b@sha256:....
+
+        If list of images is empty or "spec" section is not found, return contents as-is
+
+        :param string contents: File contents that potentially contains a "spec" section
+        :param list images: List of image info dictionaries (format: [{"name": "...", "image": "..."}])
+        :return string: Given contents with aforementioned modifications applied
+        """
+        if not images:
+            return contents
+
+        related_images = [
+            '    - name: {}\n      image: {}'.format(name, image)
+            for name, image in images.items()
+        ]
+        related_images.sort()
+
+        return re.sub(
+            r'^spec:\n',
+            'spec:\n  relatedImages:\n{}\n'.format('\n'.join(related_images)),
+            contents,
+            flags=re.MULTILINE
+        )
+
+    @property
+    def operator_name(self):
+        return self.operator_repo_name.split('/')[-1]
+
+    @property
+    def operator_csv_config(self):
+        return self.runtime.image_map[self.operator_name].config['update-csv']
+
+    @property
+    def operator_clone_path(self):
+        return '{}/distgits/{}'.format(self.runtime.working_dir, self.operator_repo_name)
+
+    @property
+    def operator_manifests_dir(self):
+        return '{}/{}'.format(
+            self.operator_clone_path,
+            self.operator_csv_config['manifests-dir'].rstrip('/')
+        )
+
+    @property
+    def operator_brew_component(self):
+        config = self.runtime.image_map[self.operator_name].config
+
+        if 'distgit' in config and 'component' in config['distgit']:
+            return config['distgit']['component']
+
+        return '{}-container'.format(self.operator_name)
+
+    @property
+    def bundle_name(self):
+        return '{}-bundle'.format(self.operator_name)
+
+    @property
+    def bundle_repo_name(self):
+        return '{}-bundle'.format(self.operator_repo_name)
+
+    @property
+    def bundle_clone_path(self):
+        return '{}-bundle'.format(self.operator_clone_path)
+
+    @property
+    def bundle_manifests_dir(self):
+        return '{}/manifests'.format(self.bundle_clone_path)
+
+    @property
+    def bundle_brew_component(self):
+        return self.operator_brew_component.replace('-container', '-bundle-container')
+
+    @property
+    def branch(self):
+        return self.runtime.group_config.branch.format(**self.runtime.group_config.vars)
+
+    @property
+    def rhpkg_opts(self):
+        opts = self.runtime.rhpkg_config
+        if hasattr(self.runtime, 'user'):
```

Comment:
I'm not entirely sure you need to check it has this attr but in any case it doesn't work too well if it's `None`, so I'd suggest:
```suggestion
        if hasattr(self.runtime, 'user') and self.runtime.user is not None:
```

Reply:
:heavy_check_mark: